src/pipeline/mod.rs

```diff
@@ -46,15 +90,37 @@ impl Pipeline {
     }
 
     pub async fn execute_with_session_context(&mut self, ctx: &mut SessionContext) -> Result<()> {
-        let table_provider = self.input_strategy.as_table_provider(ctx).await?;
+        let input_strategy = self
+            .input_strategy
+            .as_ref()
+            .ok_or_else(|| anyhow!("No input strategy provided"))?;
+
+        let output_strategy = self
+            .output_strategy
+            .as_mut()
+            .ok_or_else(|| anyhow!("No output strategy provided"))?;
+
+        if self.operations.is_empty() {
+            let stream = input_strategy.as_stream(ctx).await?;
+            output_strategy.write_stream(stream).await?;
+            return Ok(());
+        }
```

Comment:
here's the only streaming case. however, when I implement partitioning we'll need to add a step that inserts a sort at the end of the operations above this step when we need to do partitioning.